qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -287,29 +288,36 @@ def job_limit(self) -> BackendJobLimit:
                 maximum_jobs = backend_job_limit.maximum_jobs
                 running_jobs = backend_job_limit.running_jobs
 
-            * If ``maximum_jobs`` or ``running_jobs`` are ``None``, the
-                job limit information is currently not available.
-            * If ``maximum_jobs`` is equal to ``-1``, then there are
+            * If ``maximum_jobs`` is equal to ``None``, then there are
                 no limits to the maximum number of concurrent jobs a user
                 could submit to the backend at a time.
 
         Returns:
             the job limit for the backend with this provider.
 
         Raises:
-            LookupError: If jobs limit for the backend can't be found.
+            IBMQBackendApiProtocolError: If an unexpected value received from the server.
+            IBMQBackendJobLimitDataNotAvailable: If the job limit information is not available.
         """
         api_job_limit = self._api.backend_job_limit(self.name())
 
         try:
-            return BackendJobLimit.from_dict(api_job_limit)
+            job_limit = BackendJobLimit.from_dict(api_job_limit)
+            if job_limit.maximum_jobs is None or job_limit.running_jobs is None:
+                raise IBMQBackendJobLimitDataNotAvailable(
```

Comment:
while we should avoid making fields in schemas as required if possible, in case of future api changes, in this case it's simpler to just make them required. Since if the fields do change in the future, this code will raise an error anyway.

Reply:
Yeah, that would definitely simplify things! I'll go ahead and add that to the PR.